### ISSUE TITLE
Example agent implementation of OnRemoteConfig leads to infinite reconfigure loop

### ISSUE THREAD

**Issue by andykellr** [open-telemetry/opamp-go#49]
It appears that the expectation of `OnRemoteConfig` is that it return `nil` for `*protobufs.EffectiveConfig` if there is no change applied. However, the example agent always returns the effective config unless there was an error applying it:

https://github.com/open-telemetry/opamp-go/blob/e32b6163128d74cef2b4fb1af38dc1c81876460c/internal/examples/agent/agent/agent.go#L185-L194

In the implementation of the client Receiver, it sets `reportStatus` to true if an effective config is returned:

https://github.com/open-telemetry/opamp-go/blob/e32b6163128d74cef2b4fb1af38dc1c81876460c/client/internal/receiver.go#L80-L91

This results in a new status report being sent to the server. This contradicts the spec here:

https://github.com/open-telemetry/opamp-spec/blob/279e064d5461726ec0f70dfed314d5c354384584/specification.md?plain=1#L499-L502

I resolved this in my own implementation by comparing the config hash and returning nil for the effective config.

```go
	if bytes.Equal(config.GetConfigHash(), agent.effectiveConfigHash) {
		return nil, nil
	}
```

If returning `nil` for the `*protobufs.EffectiveConfig` is the desired approach of stopping the reconfigure messages, the callback documentation for `OnRemoteConfig` should be updated here:

https://github.com/open-telemetry/opamp-go/blob/e32b6163128d74cef2b4fb1af38dc1c81876460c/client/types/callbacks.go#L31-L49

Also, I was curious if the Sensu team ran into this in their implementation and noticed that they always return `nil` for the effective config in the OnRemoteConfig handler:

https://github.com/SumoLogic/sumologic-otel-collector/blob/fd9bd85c3d7a7bd6807dee4d5a5738b5c271b936/pkg/extension/opampextension/extension.go#L165

**Comment by tigrannajaryan** [comment]
Should be fixed by https://github.com/open-telemetry/opamp-go/pull/51

**Comment by andykellr** [comment]
> Should be fixed by #51

Confirmed fixed. Thanks!
